RecoTracker/DeDx/plugins/SiPixelTrackProbQXYProducer.cc

```diff
@@ -0,0 +1,187 @@
+// -*- C++ -*-
+//
+// Package:    SiPixelTrackProbQXYProducer
+// Class:      SiPixelTrackProbQXYProducer
+//
+/**\class SiPixelTrackProbQXYProducer  SiPixelTrackProbQXYProducer.cc RecoTracker/DeDx/plugins/SiPixelTrackProbQXYProducer.cc
+
+ Description: SiPixel Charge and shape probabilities combined for tracks
+
+*/
+//
+// Original Author:  Tamas Almos Vami
+//         Created:  Mon Nov 17 14:09:02 CEST 2021
+//
+
+// system include files
+#include <memory>
+#include <cmath>
+
+#include "FWCore/Framework/interface/Frameworkfwd.h"
+#include "FWCore/Framework/interface/stream/EDProducer.h"
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/Framework/interface/MakerMacros.h"
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "FWCore/Framework/interface/ESHandle.h"
+#include "FWCore/Framework/interface/ConsumesCollector.h"
+#include "FWCore/Utilities/interface/EDGetToken.h"
+
+//#include "Geometry/TrackerGeometryBuilder/interface/TrackerGeometry.h"
+//#include "Geometry/TrackerGeometryBuilder/interface/StripGeomDetUnit.h"
+#include "Geometry/CommonDetUnit/interface/PixelGeomDetUnit.h"
+
+#include "DataFormats/Common/interface/ValueMap.h"
+#include "DataFormats/TrackReco/interface/Track.h"
+#include "TrackingTools/PatternTools/interface/TrajTrackAssociation.h"
+#include "TrackingTools/TransientTrackingRecHit/interface/TransientTrackingRecHit.h"
+#include "DataFormats/TrackerRecHit2D/interface/SiPixelRecHit.h"
+#include "DataFormats/TrackReco/interface/SiPixelTrackProbQXY.h"
+
+#include "DataFormats/TrackerCommon/interface/TrackerTopology.h"
+#include "Geometry/Records/interface/TrackerTopologyRcd.h"
+
+//
+// class declaration
+//
+
+class SiPixelTrackProbQXYProducer : public edm::stream::EDProducer<> {
+public:
+  explicit SiPixelTrackProbQXYProducer(const edm::ParameterSet&);
+  ~SiPixelTrackProbQXYProducer() override = default;
+
+private:
+  void produce(edm::Event&, const edm::EventSetup&) override;
+  int factorial(int);
+
+  // ----------member data ---------------------------
+  edm::ESGetToken<TrackerTopology, TrackerTopologyRcd> tTopoToken_;
```

Comment:
```suggestion
  const edm::ESGetToken<TrackerTopology, TrackerTopologyRcd> tTopoToken_;
```
also the other one could be, if it is included in the constructor initializer list.

Reply:
ok I moved the track token to the const init list and made both of them a `const`. But this is not something I can do in 10_6_X, right?